### ISSUE TITLE
Not able to generate correct .trace file during performance testing

### ISSUE THREAD

**Issue by akhand-pratap2807** [appium/appium#11154]
## The problem
I am trying to measure the performance of an iOS app and for that, I am using appium. Unfortunately, not able to generate the correct trace file. When I am opening the generated trace file It shows the duration of the run as per my script and "Unknown" word is written in  Life cycle.  But not declared any expceted data related to CPU usage and others. 

## Environment

* Appium version (or git revision) that exhibits the issue: v1.8.0
* Last Appium version that did not exhibit the issue (if applicable): First time using
* Desktop OS/version used to run Appium: 
* Node.js version (unless using Appium.app|exe): v10.6.0
* Mobile platform/version under test: 11.2
* Real device or emulator/simulator: Real(iPad Mini4)
* Appium CLI or Appium.app|exe:Appium CLI

## Details
I saw the logs and instrument command was there:
 Starting 'instruments' with arguments: -w 1294c855817531f995547e171976f3bbc0adcca2 -t Time Profiler -D /var/folders/_k/dn0b5v1947g9rkbgk88nk6px_9pm1c/T/201876-11516-197kgop.87ydf/appium_perf_Time_Profiler_1533549798759.trace -l 120000 -p 16570

Also, I am getting line (node:12229) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead

## Link to Appium logs
https://gist.github.com/akhand-pratap2807/ed794b15d79f3457d5fed4ffaf3e7546/edit

## Code To Reproduce Issue [ Good To Have ]
public void Test1() throws IOException {
		File traceZip = new File("/Users/labuser/Desktop/trace.zip");
		try {
            HashMap<String, Object> args = new HashMap<>();
            args.put("timeout", 120000);
            args.put("pid", "current");
            args.put("profileName", "Time Profiler");
            appiumDriver.executeScript("mobile: startPerfRecord", args);
            doWorkflow(2);
            args = new HashMap<>();
            args.put("profileName", "Time Profiler");
            String b64Zip = (String)appiumDriver.executeScript("mobile: stopPerfRecord", args);
           
            byte[] bytesZip = Base64.getMimeDecoder().decode(b64Zip);
            FileOutputStream stream = new FileOutputStream(traceZip);
            stream.write(bytesZip);
            stream.close();
        } finally {
        	appiumDriver.quit();
        }
	}

private void doWorkflow(){
//Workflow of test
}

**Comment by mykola-mokhnach** [comment]
Do you get the same output if the command above is executed manually? If yes then there is nothing that can be changed in Appium.

Also, buffer issue with node10 is addressed in appium@beta

**Comment by akhand-pratap2807** [comment]
Thanks, @mykola-mokhnach. I am not able to get data while using Instruments manually as well. Seems like Xcode issues.

Although I tried to get the Memory leaks using Profile "Leaks" with both manually and using Appium.
Just by replacing the String "Time Profiler" with "Leaks". 
args.put("profileName", "Time Profiler");

I can see memory Leaks while doing the manual recording in instruments but surprised that not able to generate any data in a trace file while performing with appium server.

Thanks for helping

**Comment by mykola-mokhnach** [comment]
Did you try it without providing `pid` ?

**Comment by akhand-pratap2807** [comment]
In that case below errors are present in the .trace file:
    1> Allocations cannot handle a target type of 'All Processes'
    2> Allocations cannot handle a target type of 'All Processes'

**Comment by akhand-pratap2807** [comment]
@klementynka Could you please have a look here. Thanks

**Comment by klementynka** [comment]
Hi @akhand-pratap2807, 

Which computer and Xcode version do you use ? I'm working on MacBook Pro, Xcode 9.4.1 and I have the same problems to run such tests on a real device since 2 months when I get new laptop.

I registered a bug ticket at Apple, I was not to able run manually any of instruments with real device because Instruments crashed immediately. They closed my ticket and told it was fixed in Xcode 10.4 beta. It is right - something was fixed, because now Instruments are not crashing but now I get error message. I registered the second bug ticket and still waiting for response. 

Here I have the same issue like you have on both Xcode 9.4.1 and 10.4 beta. You are right - Time Profiler problem is for Apple. Sorry I cannot help with Leaks because at this moment I am not to able run any of this kind of tests on my real device.

**Comment by akhand-pratap2807** [comment]
Thanks, @klementynka for such a big and useful explanation.
I am using Xcode 9.4.1 as well. 
Could you please specify the last version of Xcode on which you have successfully executed the tests based on instruments?

**Comment by mykola-mokhnach** [comment]
@akhand-pratap2807 instruments might require root privileges in some cases, for example if you are trying to collect perf data from a particular process on Simulator. This is not supported by Appium, since its process is executed from non-privileged user account.

**Comment by klementynka** [comment]
@akhand-pratap2807  I'm pretty sure iOS version has no impact on crashing the instruments. On my previous laptop that I no longer have I was able to run automated tests on several iOS versions - starting from 10 and above. Since I got a new MacBook Pro I wasn't able to get it working with real devices. I'm sorry I don't remember exact versions of OS and Xcode on my old MacBook but I think they were relatively new. I really hope Apple support can resolve it soon.
